modules/nf-core/bascule/templates/main_script.R

```diff
@@ -0,0 +1,125 @@
+  parsed_args[! is.na(parsed_args)]
+}
+
+opt = list(
+  prefix = ifelse('$task.ext.prefix' == 'null', '$meta.id', '$task.ext.prefix')
+)
+args_opt = parse_args('$task.ext.args')
+for ( ao in names(args_opt)) opt[[ao]] = args_opt[[ao]]
+
+# Script
+library(bascule)
+library(ggplot2)
+library(reticulate)
+library(tidyverse)
+
+untar("$counts_matrices", verbose=TRUE)
+counts_folder = gsub(".tar.gz", "", "$counts_matrices")
+
+counts_all = lapply(list.files(counts_folder, full.names=TRUE), function(file_id) {
+  read.csv(file_id, sep="\t") %>% column_to_rownames(var="MutationType") %>%
+    t() %>% as.data.frame()
+}) %>% setNames(list.files(counts_folder))
+
+n_types = length(counts_all)
+types_names = names(counts_all)
+
+# keep only samples present in all events types
+samples_list = lapply(types_names, function(i) {
+  tibble(counts_id=i, samples=rownames(counts_all[[i]]))
+}) %>% bind_rows() %>%
+  group_by(samples) %>%
+  summarise(n_types=n()) %>%
+  filter(n_types == !!n_types) %>%
+  pull(samples) %>% as.character()
+
+counts = lapply(counts_all, function(count_i) count_i[samples_list,])
+
+reference_cat = eval(parse(text=opt[["reference_cat"]]))
+
+# reorder columns
+counts = lapply(types_names, function(type_id) {
+  if (!is.null(reference_cat[[type_id]]))
+    counts[[type_id]][,colnames(reference_cat[[type_id]])]
+}) %>% setNames(types_names)
+
+
+# set up reticulate
+## define micromamba binary as main conda binary
+options(reticulate.conda_binary=system("which micromamba", intern=TRUE))
+## define the python binary - not necessary if RETICULATE_PYTHON is defined
+# use_python(system("which python", intern=TRUE), required=TRUE)
+## load pybascule package
+py = import("pybascule")
+
+x = fit(counts=counts,
+        k_list=eval(parse(text=opt[["k_list"]])),
+        reference_cat=reference_cat,
+        keep_sigs=eval(parse(text=opt[["keep_sigs"]])),
+        store_fits=TRUE,
+        py=py)
+
+x = refine_denovo_signatures(x)
+
+clustering = FALSE
+if (length(samples_list) > 1) {
+  clustering = TRUE
+  x = fit_clustering(x, cluster=as.integer(opt[["cluster"]]))
+  x = merge_clusters(x, cutoff=as.numeric(opt[["cutoff"]]))
+}
+
+x = convert_dn_names(x, cutoff=as.numeric(opt[["cutoff"]]))
+
+# plots
+pl_signatures = plot_signatures(x)
+pl_exposures = plot_exposures(x)
+pl_centroids = plot_centroids(x)
+
+if (clustering) {
+  figure = patchwork::wrap_plots(pl_signatures,
+                                 pl_exposures + theme(legend.position="bottom"),
+                                 pl_centroids + theme(legend.position="bottom"),
+                                 design="aaaa\nbbbc")
+} else {
+  figure = patchwork::wrap_plots(pl_signatures,
+                                 pl_exposures + theme(legend.position="bottom"),
+                                 ncol=1)
+}
+
+# save objects
+saveRDS(x, file=paste0(opt[["prefix"]], "_fit_bascule.rds"))
```

Comment:
I would suggest putting prefix into the main.nf, with a reasonable default. You can incorporate it by using $, see an example in https://github.com/nf-core/modules/blob/efc7b2793391bb02bef37fb55d52e41b0b8b8839/modules/nf-core/gprofiler2/gost/templates/gprofiler2_gost.R#L137

Reply:
Sorry, I meant this link https://nf-co.re/docs/guidelines/components/modules#optional-command-arguments